src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -595,15 +596,27 @@ private void WarningRaised(object sender, BuildWarningEventArgs e)
                 threadId: e.ThreadId,
                 logOutputProperties: null);
 
-            if (_immediateMessageMarkers.Any(marker => message.IndexOf(marker, StringComparison.Ordinal) > 0))
+            if (ImmeidateMessageRaised(message))
             {
-                RenderImmediateMessage(message, MessageSeverity.Warning);
+                RenderImmediateMessage(message);
             }
 
             project.AddBuildMessage(MessageSeverity.Warning, message);
         }
     }
 
+    /// <summary>
+    /// Detectw markers that require special attention from a customer.
+    /// </summary>
+    /// <param name="message">Raised event</param>
+    /// <returns>true if marker is detected.</returns>
+    private bool ImmeidateMessageRaised(string message)
+    {
+        Regex regex = new Regex($"({_immediateMessagePattern})");
```

Comment:
we shouldn't re-create this Regex for each call to this method - we should precompute it once as a class-local value and reuse that precomputed value here.